src/lib/y2partitioner/sequences/filesystem_controller.rb

```diff
@@ -0,0 +1,316 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "y2partitioner/device_graphs"
+require "y2storage/filesystems/btrfs"
+require "y2storage/subvol_specification"
+
+module Y2Partitioner
+  module Sequences
+    # This class stores information about a filesystem being created or modified
+    # and takes care of updating the devicegraph when needed, so the different
+    # dialogs can always work directly on a BlkFilesystem object correctly
+    # placed in the devicegraph.
+    class FilesystemController
+      # @return [Symbol]
+      attr_accessor :role
+
+      # @return [Boolean] whether the user wants to encrypt the device
+      attr_accessor :encrypt
+
+      # @return [String] password for the encryption device
+      attr_accessor :encrypt_password
+
+      # return [String]
+      attr_reader :blk_device_name
+
+      DEFAULT_FS = Y2Storage::Filesystems::Type::BTRFS
+      DEFAULT_HOME_FS = Y2Storage::Filesystems::Type::XFS
+      DEFAULT_PARTITION_ID = Y2Storage::PartitionId::LINUX
+      private_constant :DEFAULT_FS, :DEFAULT_HOME_FS, :DEFAULT_PARTITION_ID
+
+      def initialize(device)
+        @blk_device_name = device.name
+        @encrypt = blk_device.encrypted?
+        @initial_graph = working_graph.dup
+      end
+
+      def blk_device
+        Y2Storage::BlkDevice.find_by_name(working_graph, blk_device_name)
+      end
+
+      def filesystem
+        blk_device.filesystem
+      end
+
+      def filesystem_type
+        filesystem ? filesystem.type : nil
+      end
+
+      def to_be_formatted?
+        return false if filesystem.nil?
+        new?(filesystem)
+      end
+
+      def to_be_encrypted?
+        return false unless can_change_encrypt?
+        encrypt && !blk_device.encrypted?
+      end
+
+      def mount_point
+        filesystem ? filesystem.mountpoint : nil
+      end
+
+      def partition_id
+        blk_device.is?(:partition) ? blk_device.id : nil
+      end
+
+      def apply_role
```

Comment:
NP: In general miss more methods documentation in this class but in the rest of classes it is well documented

Reply:
Oops, this was overlooked. Please check the new commit including your fixes and the documentation for this class.

Reply:
Thnx!!!